src/components/PresenceMap/component.js

```diff
@@ -0,0 +1,141 @@
+import React, { useRef, useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import getMapContainerStyles from './getMapContainerStyles';
+import StyledContainer from './styled/container';
+import StyledMapContainer from './styled/mapContainer';
+
+export default function MapComponent(props) {
+    const container = useRef();
+    const mapContainer = useRef();
+    const [markers, setMarker] = useState([]);
+    const MAX_ZOOM = 14;
+
+    const prevIsScriptLoaded = useRef(props.isScriptLoaded);
+    useEffect(() => {
+        if (!prevIsScriptLoaded.current && props.isScriptLoaded && props.isScriptLoadSucceed) {
+            // eslint-disable-next-line
+            initMap();
+        }
+        // eslint-disable-next-line
+    }, [props.isScriptLoaded, props.isScriptLoadSucceed]);
+
+    const getContainerClassNames = () => {
+        const { className } = props;
+        if (className && typeof className === 'string') {
+            return `rainbow-google-map ${className}`;
+        }
+        return 'rainbow-google-map';
+    };
+
+    const initMap = () => {
+        const { zoom, center, objects, showTraffic, showTransit } = props;
+
+        const mapOptions = {
+            zoom: zoom === 'auto' ? MAX_ZOOM : zoom,
+            center: center !== 'auto' ? center : null,
+            disableDefaultUI: true,
+        };
+
+        const map = new window.google.maps.Map(mapContainer.current, mapOptions);
+        const markersCopy = [...markers];
+
+        objects.forEach(marker => {
+            const markerMap = new window.google.maps.Marker({
+                position: marker.position,
+                map,
+                zIndex: Math.round(marker.position.lat * 100000),
+            });
+            markersCopy.push(markerMap);
+        });
+
+        if (center === 'auto') {
+            const bounds = new window.google.maps.LatLngBounds();
+
+            markersCopy.map(marker => bounds.extend(marker.getPosition()));
+
+            map.setCenter(bounds.getCenter());
+            map.fitBounds(bounds);
+
+            let removeListener = null;
+            const listener = window.google.maps.event.addListener(map, 'bounds_changed', () => {
```

Comment:
this listener is inside of a function `initMap()` then every time that function gets invoke you are going to attach the listener

I think this logic should happen within a useEffect for props.center with a condition of === 'auto', recall all the properties of the component can change an as a consequence the component needs to react